velox/dwio/dwrf/writer/Writer.cpp

```diff
@@ -802,6 +802,19 @@ dwrf::WriterOptions getDwrfOptions(const dwio::common::WriterOptions& options) {
   dwrfOptions.memoryPool = options.memoryPool;
   dwrfOptions.spillConfig = options.spillConfig;
   dwrfOptions.nonReclaimableSection = options.nonReclaimableSection;
+
+  if (options.flushPolicyFactory) {
+    if (auto flushPolicy = options.flushPolicyFactory()) {
+      if (auto dwrfFlushPolicy =
```

Comment:
This logic suggests that if policy specified in the options is of the "wrong" type, then it is silently ignored. Is this intentional? It might be better to throw.

Reply:
@mbasmanova Yes. If the options is wrong type, It will be set the DefaultFlushPolicy [here](https://github.com/facebookincubator/velox/blob/main/velox/dwio/dwrf/writer/Writer.cpp#L87).

Reply:
It seems error prone to ignore "invalid" policy. Consider throwing instead.